bump.go

```diff
@@ -0,0 +1,103 @@
+package paymail
+
+import (
+	"errors"
+	"github.com/libsv/go-bc"
+)
+
+// BUMPPaths represents BUMP format for all inputs
+type BUMPPaths []BUMP
+
+// BUMP is a struct that represents a whole BUMP format
+type BUMP struct {
+	blockHeight uint64
+	path        []BUMPPath
+}
+
+// BUMPPath is a slice of BUMPLevel objects which represents a path
+type BUMPPath []BUMPPathElement
+
+// BUMPPathElement is a struct that represents a single BUMP transaction
```

Comment:
"Leaf" is the accurate term to use here instead of "transaction".

"BUMPPath" is a stutter because the P in BUMP is "Path" ==> BUMPs I think would be better. Perhaps even drop the plural and simply use a slice of BUMP? I don't know what the Go convention is on that.

```go
type BUMP struct {
	blockHeight uint64
	path        []path
}
```

On level 0 of the Merkle tree the leaves are txids - but the bool txId is just supposed to indicate whether or not the hash is of interest to the user or whether it just happens to be the hash which is a sibling of the txid the user is interested in.

It's possible that _both_ siblings are of interest in which case both would have txId = true.

```
       o
   o      o 
o  o    o  o
```
Each `o` above is a leaf in the Merkle tree, but only the bottom level are txids. We may use the format to encapsulate partial paths also - for example if we split a large merkle tree across shards, we may need to be able to encode the top part separate from the bottom.

Reply:
Sure, thanks for explanation! We changed structs here and in BUX